internal/factory/deployment.go

```diff
@@ -0,0 +1,50 @@
+package factory
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// NewDeployment stubs an instance of a deployment
+func NewDeployment(deploymentName, namespace, loggingComponent, component, impl string, podSpec core.PodSpec, visitors ...func(o runtime.Object)) *apps.Deployment {
```

Comment:
deploymentName and Namespace should be swapped for 2 reasons:

* prefer namespace/name in general for args
* to match the actual [caller](https://github.com/openshift/cluster-logging-operator/pull/2240/files#diff-f143e28681fca473048e426218fd5def9528a1dd9c5279be226665a3ab1a5601R23)